Samba:s3:modules: fix aio_fsync function in vfs_aio_fbsd

anodos325 opened the pull request:
Set aiocb state to in_progress so that we properly set the
request to done and call the fsync handler function.

yocalebo reviewed (approved):
I trust you tested 👍

anodos325 committed:
Samba:s3:modules: fix aio_fsync function in vfs_aio_fbsd

Set aiocb state to in_progress so that we properly set the
request to done and call the fsync handler function.
